pkg/leeway/signing/attestation_test.go

```diff
@@ -1206,3 +1211,420 @@ func TestFetchGitHubOIDCToken(t *testing.T) {
+			},
+			expectError: true,
+			errorMsg:    "sub claim not found or empty",
+		},
+		{
+			name: "job_workflow_ref in top-level claim (not in sub)",
+			setupServer: func() *httptest.Server {
+				return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					header := base64EncodeForTest(`{"alg":"RS256","typ":"JWT"}`)
+					payload := base64EncodeForTest(`{
+						"sub": "repo:org/repo:environment:prod",
+						"aud": "sigstore",
+						"job_workflow_ref": "org/repo/.github/workflows/deploy.yml@refs/heads/main"
+					}`)
+					signature := base64EncodeForTest("fake-signature")
+					token := fmt.Sprintf("%s.%s.%s", header, payload, signature)
+					
+					w.Header().Set("Content-Type", "application/json")
+					if err := json.NewEncoder(w).Encode(map[string]string{"value": token}); err != nil {
+						t.Errorf("Failed to encode response: %v", err)
+					}
+				}))
+			},
+			githubCtx: &GitHubContext{
+				ServerURL:  "https://github.com",
+				Repository: "org/repo",
+			},
+			expectError: false,
+			expectedID:  "https://github.com/org/repo/.github/workflows/deploy.yml@refs/heads/main",
+		},
+		{
+			name: "missing job_workflow_ref in sub claim and top-level",
+			setupServer: func() *httptest.Server {
+				return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					header := base64EncodeForTest(`{"alg":"RS256","typ":"JWT"}`)
+					payload := base64EncodeForTest(`{
+						"sub": "repo:org/repo:ref:refs/heads/main",
+						"aud": "sigstore"
+					}`)
+					signature := base64EncodeForTest("fake-signature")
+					token := fmt.Sprintf("%s.%s.%s", header, payload, signature)
+					
+					w.Header().Set("Content-Type", "application/json")
+					if err := json.NewEncoder(w).Encode(map[string]string{"value": token}); err != nil {
+						t.Errorf("Failed to encode response: %v", err)
+					}
+				}))
+			},
+			githubCtx: &GitHubContext{
+				ServerURL:  "https://github.com",
+				Repository: "org/repo",
+			},
+			expectError: true,
+			errorMsg:    "job_workflow_ref not found",
+		},
+		{
+			name: "OIDC token fetch failure",
+			setupServer: func() *httptest.Server {
+				return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					w.WriteHeader(http.StatusUnauthorized)
+				}))
+			},
+			githubCtx: &GitHubContext{
+				ServerURL:  "https://github.com",
+				Repository: "org/repo",
+			},
+			expectError: true,
+			errorMsg:    "failed to fetch OIDC token",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			server := tt.setupServer()
+			defer server.Close()
+
+			oldRequestURL := os.Getenv("ACTIONS_ID_TOKEN_REQUEST_URL")
+			oldRequestToken := os.Getenv("ACTIONS_ID_TOKEN_REQUEST_TOKEN")
+			defer func() {
+				os.Setenv("ACTIONS_ID_TOKEN_REQUEST_URL", oldRequestURL)
+				os.Setenv("ACTIONS_ID_TOKEN_REQUEST_TOKEN", oldRequestToken)
+			}()
+
+			os.Setenv("ACTIONS_ID_TOKEN_REQUEST_URL", server.URL)
```

Comment:
Nit:
Can usage of `os.SetEnv` be replaced with `t.Setenv`?

Reply:
On it